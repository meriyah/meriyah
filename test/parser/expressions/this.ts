import { describe } from 'vitest';
import { pass } from '../../test-utils';

describe('Expressions - This', () => {
  pass('Expressions - This (pass)', ['this._global = _global;', 'this\n/foo;', 'this\n/foo/g;']);
});
