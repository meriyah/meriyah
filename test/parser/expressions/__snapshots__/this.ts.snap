// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Expressions - This > Expressions - This (pass) > this
/foo/g; 1`] = `
{
  "body": [
    {
      "expression": {
        "left": {
          "left": {
            "type": "ThisExpression",
          },
          "operator": "/",
          "right": {
            "name": "foo",
            "type": "Identifier",
          },
          "type": "BinaryExpression",
        },
        "operator": "/",
        "right": {
          "name": "g",
          "type": "Identifier",
        },
        "type": "BinaryExpression",
      },
      "type": "ExpressionStatement",
    },
  ],
  "sourceType": "script",
  "type": "Program",
}
`;

exports[`Expressions - This > Expressions - This (pass) > this
/foo; 1`] = `
{
  "body": [
    {
      "expression": {
        "left": {
          "type": "ThisExpression",
        },
        "operator": "/",
        "right": {
          "name": "foo",
          "type": "Identifier",
        },
        "type": "BinaryExpression",
      },
      "type": "ExpressionStatement",
    },
  ],
  "sourceType": "script",
  "type": "Program",
}
`;

exports[`Expressions - This > Expressions - This (pass) > this._global = _global; 1`] = `
{
  "body": [
    {
      "expression": {
        "left": {
          "computed": false,
          "object": {
            "type": "ThisExpression",
          },
          "optional": false,
          "property": {
            "name": "_global",
            "type": "Identifier",
          },
          "type": "MemberExpression",
        },
        "operator": "=",
        "right": {
          "name": "_global",
          "type": "Identifier",
        },
        "type": "AssignmentExpression",
      },
      "type": "ExpressionStatement",
    },
  ],
  "sourceType": "script",
  "type": "Program",
}
`;
